testing-internal/src/main/java/com/linecorp/armeria/internal/testing/DynamicBehaviorHandler.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.testing;
+
+import javax.annotation.Nullable;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+
+@Sharable
+public class DynamicBehaviorHandler extends ChannelDuplexHandler {
```

Comment:
This seems unused but I guess you have a plan. :wink:

Reply:
It's used to control proxy server behavior while running integration tests. 🏃 

https://github.com/line/armeria/blob/5c79ac09cb59f0e6e463b32ffe4cd2e2f362010c/core/src/test/java/com/linecorp/armeria/client/proxy/ProxyClientIntegrationTest.java#L94